SU2_CFD/src/output/filewriter/CSTLFileWriter.cpp

```diff
@@ -0,0 +1,332 @@
+   partitions so we can set up buffers. The master node will handle
+   the writing of the STL file after gathering all of the data. ---*/
+
+  unsigned long nLocalTria = dataSorter->GetnElem(TRIANGLE),
+                nLocalQuad = dataSorter->GetnElem(QUADRILATERAL),
+                nLocalTriaAll = nLocalTria + nLocalQuad*2; // Quad splitted into 2 tris
+
+  if (rank == MASTER_NODE) buffRecvTriaCount = new unsigned long[nProcessor];
+
+  /*--- Communicate the maximum of local triangles on any process to each partition and the number of local vertices on each partition
+   to the master node with collective calls. ---*/
+
+  SU2_MPI::Allreduce(&nLocalTriaAll, &max_nLocalTriaAll, 1,
+                     MPI_UNSIGNED_LONG, MPI_MAX, MPI_COMM_WORLD);
+
+
+  SU2_MPI::Gather(&nLocalTriaAll   , 1, MPI_UNSIGNED_LONG,
+                  buffRecvTriaCount, 1, MPI_UNSIGNED_LONG,
+                  MASTER_NODE, MPI_COMM_WORLD);
+
+  /*--- Allocate buffer for send/recv of the coordinate data. Only the master rank allocates buffers for the recv. ---*/
+  buffSendCoords = new su2double[max_nLocalTriaAll*N_POINTS_TRIANGLE*3]; /* Triangle has 3 Points with 3 coords each */
+  if (rank == MASTER_NODE)
+    buffRecvCoords = new su2double[nProcessor*max_nLocalTriaAll*N_POINTS_TRIANGLE*3];
+
+  StoreCoordData(TRIANGLE,      nLocalTria, 0ul);
+  StoreCoordData(QUADRILATERAL, nLocalQuad, nLocalTria*N_POINTS_TRIANGLE*3);
+
+  /*--- Collective comms of the solution data and global IDs. ---*/
+  SU2_MPI::Gather(buffSendCoords, static_cast<int>(max_nLocalTriaAll*N_POINTS_TRIANGLE*3), MPI_DOUBLE,
+                  buffRecvCoords, static_cast<int>(max_nLocalTriaAll*N_POINTS_TRIANGLE*3), MPI_DOUBLE,
+                  MASTER_NODE, MPI_COMM_WORLD);
+
+  /*--- Free temporary memory. ---*/
+  if(buffSendCoords != NULL) delete [] buffSendCoords;
+}
+
+
+void CSTLFileWriter::StoreCoordData(enum GEO_TYPE elemType,
+                                    unsigned long nLocalElements,
+                                    unsigned long startIndex) {
+
+  unsigned long globalNodeNumber, localNodeNumber, iElem, index = startIndex;
+  unsigned short iPoint, iVar;
+
+  /*--- Create a node list that defines in which order the element points form a triangle. ---*/
+  vector<unsigned short> nodeList;
+  switch (elemType) {
+    case TRIANGLE:
+      nodeList.insert(nodeList.end(), {0,1,2} ); break;
+    case QUADRILATERAL:
+      /*--- Quads split into two Triangles. Assumes either clock- or counterclockwise rotation ---*/
+      nodeList.insert(nodeList.end(), {0,1,3, 1,2,3} ); break;
+    default:
+      SU2_MPI::Error("Unsupported element type for the STL writer." , CURRENT_FUNCTION); break;
+  }
+
+  /*--- Write Triangle-Point-Coordinates subsequently in a local send buffer. ---*/
+  for (iElem = 0; iElem < nLocalElements; iElem++) {
+    for (auto iPoint : nodeList) {
+      globalNodeNumber = dataSorter->GetElem_Connectivity(elemType, iElem, nodeList[iPoint]) - 1;

```

Comment:
By doing iPoint : nodeList, iPoint is the contents of nodeList, i.e. instead of nodeList[iPoint] you just use iPoint.

Reply:
This is probably why the test failed.

Reply:
yep, fixed